server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1528,6 +1528,145 @@ public int countConsumers(
         return poolDTOs;
     }
 
+
+    /**
+     * Retrieves a list of Pools ordered by Product for an Owner
+     *
+     * @param ownerKey id of the owner whose entitlement pools are sought.
+     * @param matches Find pools matching the given pattern in a variety of fields.
+     * * and ? wildcards are supported.
+     * @return a list of Pool objects
+     * @httpcode 400
+     * @httpcode 404
+     * @httpcode 200
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{owner_key}/pools_ordered_by_product")
+    @SuppressWarnings("checkstyle:indentation")
+    @ApiOperation(notes = "Retrieves a list of Pools ordered by Product for an Owner", value = "List Pools")
+    @ApiResponses({
+            @ApiResponse(code = 404, message = "Owner not found"),
+            @ApiResponse(code = 400, message = "Invalid request")
+    })
+    public List<PoolDTO> listPoolsOrderedByProduct(
```

Comment:
With the addition of the product alias, is this something where we need another end point, or does passing in "Product.name" in as the argument for the "sort_by" parameter on the base getPools end point work as one would expect?

Reply:
We need the alias for the new end point to work

Reply:
But we do not need a new endpoint for the functionality. Once the product alias is present in the criteria, we can use the existing paging infrastructure to provide the sorting functionality without needing a new endpoint.

For example, the following curl command demonstrates this in your branch:
```curl -k -u admin:admin "https://localhost:8443/candlepin/owners/admin/pools?include=id&include=productName&sort_by=Product.name&order=ASC"```

Whereas the same command on master results in:
```{
  "displayMessage" : "Runtime Error could not resolve property: Product of: org.candlepin.model.Pool at org.hibernate.persister.entity.AbstractPropertyMapping.propertyException:73",
  "requestUuid" : "e5e2ceb1-88fe-4db1-8a2f-b86ad78948ea"
}```

If there's a compelling reason to keep both endpoints, then at the very least this new method should simply call into the existing one after configuring the paging bits as it's doing now. But, again, if that's all it's doing, then there's really no reason to not have the users just specify the sorting as necessary.